core/modules/parsers/wikiparser/rules/syslink.js

```diff
@@ -0,0 +1,45 @@
+/*\
+title: $:/core/modules/parsers/wikiparser/rules/syslink.js
+type: application/javascript
+module-type: wikirule
+
+Wiki text inline rule for system tiddler links.
+Can be suppressed preceding them with `~`.
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+exports.name = "syslink";
+exports.types = {inline: true};
+
+exports.init = function(parser) {
+	this.parser = parser;
+	// Regexp to match
+	this.matchRegExp = /~?(?:\$):[^\s<>{}\[\]`|'"\\^~]+(?:\/|\b)/mg;
```

Comment:
hmmm: `$:/asdf{[]}` imo is a valid system tiddler name? ... at least creating it isn't prevented.

The excluded characters may be illegal with the external link regexp. But imo for system tiddlers they aren't.


full regexp description
```
// ~?(?:\$):[^\s<>{}\[\]`|'"\\^~]+(?:\/|\b)
// 
// Options: case insensitive; ^ and $ match at line breaks
// 
// Match the character “~” literally «~?»
//    Between zero and one times, as many times as possible, giving back as needed (greedy) «?»
// Match the regular expression below «(?:\$)»
//    Match the character “$” literally «\$»
// Match the character “:” literally «:»
// Match a single character NOT present in the list below «[^\s<>{}\[\]`|'"\\^~]+»
//    Between one and unlimited times, as many times as possible, giving back as needed (greedy) «+»
//    A whitespace character (spaces, tabs, and line breaks) «\s»
//    One of the characters “<>{}” «<>{}»
//    A [ character «\[»
//    A ] character «\]»
//    One of the characters “`|'"” «`|'"»
//    A \ character «\\»
//    One of the characters “^~” «^~»
// Match the regular expression below «(?:\/|\b)»
//    Match either the regular expression below (attempting the next alternative only if this one fails) «\/»
//       Match the character “/” literally «\/»
//    Or match regular expression number 2 below (the entire group fails if this one fails to match) «\b»
//       Assert position at a word boundary «\b»
```